### ISSUE TITLE
No module named six

### ISSUE THREAD

**Issue by matentzn** [INCATools/ontology-development-kit#665]
When trying to use update_repo with `odklite`, I get a confusing error:

```
ODK_IMAGE=odklite TAG_IN_IMAGE=v1.3.2 ODK_DEBUG=yes ./run.sh make update_repo
Running  with -Xmx16G of memory for ROBOT and Java-based pipeline steps.
cpont.Makefile:36: warning: overriding recipe for target 'imports/mondo_import.owl'
Makefile:314: warning: ignoring old recipe for target 'imports/mondo_import.owl'
cpont.Makefile:49: warning: overriding recipe for target 'imports/uberon_import.owl'
Makefile:328: warning: ignoring old recipe for target 'imports/uberon_import.owl'
cpont.Makefile:65: warning: overriding recipe for target 'imports/ncit_import.owl'
Makefile:332: warning: ignoring old recipe for target 'imports/ncit_import.owl'
sh ../scripts/update_repo.sh
This (experimental) update script will create a new repo according to your config file. It will:
(1) overwrite your repositories Makefile, ODK sparql queries (your custom queries wont be touched) and docker wrapper (run.sh).
(2) and add missing files, if any.
Traceback (most recent call last):
  File "/tools/odk.py", line 15, in <module>
    from dataclasses_jsonschema import JsonSchemaMixin
  File "/usr/local/lib/python3.10/dist-packages/dataclasses_jsonschema/__init__.py", line 21, in <module>
    from .field_types import (  # noqa: F401
  File "/usr/local/lib/python3.10/dist-packages/dataclasses_jsonschema/field_types.py", line 8, in <module>
    from dateutil.parser import parse
  File "/usr/local/lib/python3.10/dist-packages/dateutil/parser/__init__.py", line 2, in <module>
    from ._parser import parse, parser, parserinfo, ParserError
  File "/usr/local/lib/python3.10/dist-packages/dateutil/parser/_parser.py", line 42, in <module>
    import six
ModuleNotFoundError: No module named 'six'
make: *** [Makefile:641: update_repo] Error 1
Command exited with non-zero status 2
### DEBUG STATS ###
Elapsed time: 0:00.38
Peak memory: 14532 kb
Please remember to update your ODK image from time to time: https://oboacademy.github.io/obook/howto/odk-update/.
```

Works fine:

```
ODK_IMAGE=odkfull TAG_IN_IMAGE=v1.3.2 ODK_DEBUG=yes ./run.sh make update_repo
```

Found this:
https://bobbyhadz.com/blog/python-no-module-named-six (not sure what to make of it)

For some reason adding six to the odklite requirements txt is not sufficient..

**Comment by gouttegd** [comment]
Can’t reproduce here with my latest build dating from last week. Rebuilding now to test.

> For some reason adding six to the odklite requirements txt is not sufficient..

Have you rebuilt `odkbuild`? Remember that Python packages are installed during the build of `odkbuild`. When building `odklite`, we simply copy over the packages that have been installed in the “staging area“ in the `odkbuild` image.

As an aside:

```sh
TAG_IN_IMAGE=v1.3.2 ./run.sh make update_repo
```

This cannot work. `TAG_IN_IMAGE` is an internal variable of `run.sh`, which is _not_ intended to be overridable. If you want to specify a tag, you should use `ODK_TAG`.

Compare how these variables are defined in `run.sh`:

```sh
TAG_IN_IMAGE=$(echo $ODK_IMAGE | awk -F':' '{ print $2 }')
[…]
ODK_TAG=${ODK_TAG:-latest}
```

`ODK_TAG` is only assigned a value if it is empty (or inexistent), which makes it overridable by calling the script with `ODK_TAG=... ./run.sh`.

`TAG_IN_IMAGE` _always_ gets assigned a value, regardless of whether the variable already existed or contained anything. Calling `TAG_IN_IMAGE=... ./run.sh` has no effect — whatever value you explicitly assign to that variable is discarded.

The bottom line is that to specify a tag, use either

* `ODK_IMAGE=image:tag` to specify both the image and the tag, or
* `ODK_IMAGE=image ODK_TAG=tag` (exactly the same effect as the previous line), or
* `ODK_TAG=tag` to specify the tag alone.

Note that if `ODK_IMAGE` contains a tag, it takes precedence over `ODK_TAG`, so if you pass `ODK_IMAGE=image:tag1 ODK_TAG=tag2`, `ODK_TAG` is ignored.

**Comment by gouttegd** [comment]
OK, reproduced here with a fresh build. Got a fix on the way.
